pkg/runctl/controller.go

```diff
@@ -90,9 +96,29 @@ func NewController(factory k8s.ClientFactory, metrics metrics.Metrics) *Controll
 			controller.handleTektonTaskRun(new)
 		},
 	})
+
 	return controller
 }
 
+//meterCurrentPipelineStatus writes the current Duration of Pipeline States in a histogram
+func (c *Controller) meterCurrentPipelineStatus() {
+	klog.V(4).Infof("Current metrics Observation still alive")
+	objs := c.pipelineRunStore.List()
+	for _, obj := range objs {
+		pipelineRun, ok := obj.(*api.PipelineRun)
+		if !ok {
+			klog.V(4).Infof("failed to cast cached object of type %T to a PipelineRun", obj)
+			continue
+		}
+		if pipelineRun.Status.State != api.StateFinished {
+			err := c.metrics.ObserveCurrentDurationByState(pipelineRun)
+			if err != nil {
+				klog.V(4).Infof("metrics observation of PipelineRun %v failed: %v", pipelineRun, err)
+			}
+		}
+	}
```

Comment:
We should ignore objects which have `metadata.deletionTimestamp` set. There may be (even long) delays in deletion due to finalizers, but this metric is not made to detect finalization issues.

Assume some pipeline runs get marked as deleted in state `running` but a third party has set a finalizer which doesn't get processed. Then our histogram indicates that there are pipeline runs in `running` state for a long time, but it's not true. This may confuse operators, and they start investigating into the wrong direction.

To detect finalization issues, we should have other means. This is out of scope here.